From af39019a0d214a8e708a1bcfc13cb757d34cd176 Mon Sep 17 00:00:00 2001
From: Stuart MENEFY <stuart.menefy@st.com>
Date: Thu, 4 Jun 2009 14:17:38 +0100
Subject: [PATCH] sh_stm: Don't write to sysconfig for PIO pins which don't need it

Some PIO pins do not require sysconfig level configuration to select
which alternative function will drive the pin. This has always been
flagged in the pio_sysconf array with a syscfg field of 0. However this
caused any configuration of these pins to write to sysconf config 0,
which was incorrect. So add an explicit check for this in
stx7105_pio_sysconf() and return immediately in this case.

Signed-off-by: Stuart Menefy <stuart.menefy@st.com>
---
 arch/sh/kernel/cpu/sh4/setup-stx7105.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/arch/sh/kernel/cpu/sh4/setup-stx7105.c b/arch/sh/kernel/cpu/sh4/setup-stx7105.c
index 866bbf6..61fcba1 100644
--- a/arch/sh/kernel/cpu/sh4/setup-stx7105.c
+++ b/arch/sh/kernel/cpu/sh4/setup-stx7105.c
@@ -57,6 +57,9 @@ static void stx7105_pio_sysconf(int bank, int pin, int alt, const char* name)
 	int cfg = pio_sysconf[bank].syscfg;
 	struct sysconf_field *sc[3];
 
+	if ((cfg == 0) || alt == -1))
+		return;
+
 	sc[0] = sysconf_claim(SYS_CFG, cfg, pin, pin, name);
 	sc[1] = sysconf_claim(SYS_CFG, cfg, pin+8, pin+8, name);
 	sc[2] = sysconf_claim(SYS_CFG, cfg, pin+16, pin+16, name);
-- 
1.6.0.6

